ompi/mca/coll/han/coll_han_subcomms.c

```diff
@@ -116,7 +116,7 @@ int mca_coll_han_comm_create_new(struct ompi_communicator_t *comm,
     /*
      * This sub-communicator contains the ranks that share my node.
      */
-    opal_info_set(&comm_info, "ompi_comm_coll_preference", "^han");
+    opal_info_set(&comm_info, "ompi_comm_coll_preference", "han");
```

Comment:
Why this change? IIRC, the intention is to disable HAN in the sub communicators, not enable it.

Reply:
In the dynamic path, the decision for which submodule to use on each level is taken at collective-op-time, instead of on comm-split-time, like in the classic path. The topo_lvl key attached to the new comm will let han know that this is a sub-comm and decide accordingly when a collective is called on it

Reply:
So that fixes the dynamic selection. But doesn't that interfere with the static selection? I remember we put the preference key in in order to prevent han from selecting itself in the subcomm.

Reply:
So the two paths exist together. Some methods use the dynamic (mainly the simple ones), through `mca_coll_han_comm_create_new`, while others use the old/non-dynamic one through `mca_coll_han_comm_create`. If we exclude HAN on the dynamic path sub-comm, the component with the next higher priority gets chosen, but in this way the submodule to use cannot be tuned

Reply:
So, essentially the two paths do not affect each other. Each one creates different sub-comms, with slightly different way, and the chosen path is determined by the respective collective primitive implementation. Yes this is a bit akward! (#10438)

Reply:
Btw if you attempt to test all this I believe you will need a fix for #10335

Reply:
I see now that this is in `mca_coll_han_comm_create_new` (a terrible name, btw). I guess that's fine then